packages/story-editor/src/app/rightClickMenu/provider.js

```diff
@@ -781,20 +789,36 @@ function RightClickMenuProvider({ children }) {
 
   const pageItems = useMemo(() => {
     const disableBackgroundMediaActions = selectedElement?.isDefaultBackground;
+    const isVideo = selectedElement?.type === 'video';
+    const detachLabel = isVideo
+      ? RIGHT_CLICK_MENU_LABELS.DETACH_VIDEO_FROM_BACKGROUND
+      : RIGHT_CLICK_MENU_LABELS.DETACH_IMAGE_FROM_BACKGROUND;
+    const scaleLabel = isVideo
+      ? RIGHT_CLICK_MENU_LABELS.SCALE_AND_CROP_BACKGROUND_VIDEO
+      : RIGHT_CLICK_MENU_LABELS.SCALE_AND_CROP_BACKGROUND_IMAGE;
 
     return [
       {
-        label: RIGHT_CLICK_MENU_LABELS.DETACH_IMAGE_FROM_BACKGROUND,
+        label: detachLabel,
         onClick: handleRemoveMediaFromBackground,
         disabled: disableBackgroundMediaActions,
         ...menuItemProps,
       },
       {
-        label: RIGHT_CLICK_MENU_LABELS.SCALE_AND_CROP_BACKGROUND,
+        label: scaleLabel,
         onClick: handleOpenScaleAndCrop,
         disabled: disableBackgroundMediaActions,
         ...menuItemProps,
       },
+      ...(isVideo && hasTrimMode
```

Comment:
`hasTrimMode` is now `true` while `isTrimming` or `isMuting` etc. as well.
It would probably need `disabled: !canTranscodeResource(resource);` here to disable it for the time of any video processing.